transport/rif_comms/node.py

```diff
@@ -281,9 +248,9 @@ def send_message(self, payload: str, recipient: Address):
         is_subscribed_to_receiver_topic = self._comms_client.is_subscribed_to(recipient)
         # if not, create the topic subscription
         if not is_subscribed_to_receiver_topic:
-            topic_id, _ = self._comms_client.subscribe_to(recipient)
+            self._comms_client.subscribe_to(recipient)
         # send the message
-        self._comms_client.send_message(payload, topic_id)  # TODO: exception handling for RIF Comms client
```

Comment:
`topic_id` might be not set here, but we always have the recipient

Reply:
> i was using the topic id here because of that

we can't do this because `if not is_subscribed_to_receiver_topic == False`, then `topic_id` will raise a `NameError` exception, as it is undefined.

to be honest, i _think_ the **only** reason why this is working so far is because the `is_subscribed_to` calls are always (wrongly) returning `False`—this is a known bug—and so you always fetched the `topic_id`.

> but this assumes that the sendMessageToTopic can work with an rsk address right?
> 
> is that working?

yes, i've verified this through the tests.

note that:
1. now, `_get_peer_id` is being called inside the `send_message` function, which allows us to get the topic id.
2. `sendMessageToTopic` still uses a topic ID (same as before).

see here:
https://github.com/rsksmart/lumino/blob/a24ced2059d6512e3a47e1beba7a11cd3f143e8b/transport/rif_comms/client.py#L74-L88

---

another reason for this change is these signatures:

https://github.com/rsksmart/lumino/blob/a24ced2059d6512e3a47e1beba7a11cd3f143e8b/transport/node.py#L55-L61

https://github.com/rsksmart/lumino/blob/a24ced2059d6512e3a47e1beba7a11cd3f143e8b/transport/rif_comms/node.py#L243-L256

it seems preferable to me that the client matches these 2.

Reply:
Lgtm then